tooling/nargo_cli/src/cli/mod.rs

```diff
@@ -223,31 +222,32 @@ fn lock_workspace(
         }
     }
 
-    let mut locks = Vec::new();
-    for pkg in workspace.into_iter() {
-        let toml_path = get_package_manifest(&pkg.root_dir)?;
-        let path_display = toml_path.display();
-
-        let file = File::open(&toml_path)
-            .unwrap_or_else(|e| panic!("Expected {path_display} to exist: {e}"));
-
-        if exclusive {
-            if fs2::FileExt::try_lock_exclusive(&file).is_err() {
-                eprintln!("Waiting for lock on {path_display}...");
-            }
-            fs2::FileExt::lock_exclusive(&file)
-                .unwrap_or_else(|e| panic!("Failed to lock {path_display}: {e}"));
-        } else {
-            if fs2::FileExt::try_lock_shared(&file).is_err() {
-                eprintln!("Waiting for lock on {path_display}...",);
-            }
-            fs2::FileExt::lock_shared(&file)
-                .unwrap_or_else(|e| panic!("Failed to lock {path_display}: {e}"));
-        }
+    let lock_path = target_dir.join(".nargo-lock");
+    let path_display = lock_path.display();
+
+    std::fs::create_dir_all(target_dir).unwrap();
+    let file = OpenOptions::new()
+        .create(true)
+        .truncate(false)
+        .write(true)
+        .open(&lock_path)
+        .unwrap_or_else(|e| panic!("Expected {path_display} to exist: {e}"));
 
-        locks.push(LockedFile(file));
+    if exclusive {
+        if fs2::FileExt::try_lock_exclusive(&file).is_err() {
+            eprintln!("Waiting for lock on {path_display}...");
+        }
+        fs2::FileExt::lock_exclusive(&file)
+            .unwrap_or_else(|e| panic!("Failed to lock {path_display}: {e}"));
+    } else {
+        if fs2::FileExt::try_lock_shared(&file).is_err() {
+            eprintln!("Waiting for lock on {path_display}...",);
+        }
+        fs2::FileExt::lock_shared(&file)
+            .unwrap_or_else(|e| panic!("Failed to lock {path_display}: {e}"));
     }
-    Ok(locks)
+
+    Ok(LockedFile(file))
```

Comment:
I would also suggest removing the lock files to not leave around hundreds of them, but maybe that's not important. Or maybe they can all be in a `target/locks` directory, so nobody is bothered.